src/gwt/src/org/rstudio/studio/client/common/spelling/TypoSpellChecker.java

```diff
@@ -14,224 +14,199 @@
-      
+      uiPrefs_.realTimeSpellChecking().addValueChangeHandler(prefChangedHandler_);
+      uiPrefs_.ignoreWordsInUppercase().addValueChangeHandler(prefChangedHandler_);
+      uiPrefs_.ignoreWordsWithNumbers().addValueChangeHandler(prefChangedHandler_);
+
       // subscribe to user dictionary changes
-      context_.releaseOnDismiss(userDictionary_.addListChangedHandler(
-                                                   new ListChangedHandler() {
-         @Override
-         public void onListChanged(ListChangedEvent event)
+      context_.releaseOnDismiss(userDictionary_.addListChangedHandler((ListChangedEvent event) ->
          {
             // detect whether this is the first delivery of the list
             // or if it is an update
             boolean isUpdate = userDictionaryWords_ != null;
-                 
+
             userDictionaryWords_ = event.getList();
-            
+
             updateIgnoredWordsIndex();
-            
+
             if (isUpdate)
-               context_.invalidateMisspelledWords(); 
+               context_.invalidateMisspelledWords();
          }
-      }));
+      ));
    }
-   
+
    @Inject
-   void intialize(SpellingService spellingService,
-                  WorkbenchListManager workbenchListManager,
-                  UIPrefs uiPrefs)
+   void initialize(SpellingService spellingService, WorkbenchListManager workbenchListManager, UIPrefs uiPrefs)
    {
-      spellingService_ = spellingService;
       userDictionary_ = workbenchListManager.getUserDictionaryList();
       uiPrefs_ = uiPrefs;
+
+      ExternalJavaScriptLoader.Callback loadTypoCallback = () -> typoNative = new TypoNative("en_US");
+      new ExternalJavaScriptLoader(TypoResources.INSTANCE.typojs().getSafeUri().asString()).addCallback(loadTypoCallback);
+   }
+
+   // Check the spelling of a single word, directly returning an
+   // array of suggestions for corrections. The array is empty if the
+   // word is deemed correct by the dictionary
+   public boolean checkSpelling(String word)
+   {
+      return typoNative.check(word);
    }
-   
-   public void checkSpelling(
-                  List<String> words, 
-                  final ServerRequestCallback<SpellCheckerResult> callback)
+
+   public void checkSpelling(List<String> words, final ServerRequestCallback<SpellCheckerResult> callback)
    {
       // allocate results
       final SpellCheckerResult spellCheckerResult = new SpellCheckerResult();
 
+      // if not checking any words, return
       if (words.isEmpty())
       {
          callback.onResponseReceived(spellCheckerResult);
          return;
       }
-      
-      // only send words to the server that aren't ignored
-      final ArrayList<String> wordsToCheck = new ArrayList<String>();
-      for (int i = 0; i<words.size(); i++)
+
+      for (String word : words)
       {
-         String word = words.get(i);
+         //Debug.log("Checking word: " + word);
```

Comment:
Be sure to delete this before checkin. Convention is to use "Debug.devlog()" for this sort of thing and then be sure none of those every make it into a checkin.

Reply:
Do we not squash/rebase before we merge into the primary branch? I like to commit often and even use PRs for scratch branches to get exposure and conversation, squashing will clean the history of any debug output that was there previously. I tried to make this a "Draft PR" but it seems that functionality hasn't made it to this repo yet.

Reply:
Sorry, meant to say "before merge into primary branch" not "before commit".